libs/blocks/carousel/carousel.js

```diff
@@ -496,6 +496,41 @@ export default function init(el) {
   nextPreviousContainer.append(...nextPreviousBtns, controlsContainer);
   el.append(nextPreviousContainer);
 
+  function normalizeVideoHeights() {
+    const videos = el.querySelectorAll('video');
+    if (videos.length === 0) return;
+
+    let maxHeight = 0;
+    videos.forEach((video) => {
+      const videoHeight = video.offsetHeight;
+
+      const foreground = video.closest('.editorial-card')?.querySelector('.foreground');
+      let totalHeight = videoHeight;
+
+      if (foreground) {
+        const foregroundHeight = foreground.offsetHeight;
+        totalHeight += foregroundHeight;
+      }
+
+      if (totalHeight > maxHeight) maxHeight = totalHeight;
+    });
+
+    videos.forEach((video) => {
+      const foreground = video.closest('.editorial-card')?.querySelector('.foreground');
+
+      if (foreground) {
+        const videoHeight = video.offsetHeight;
+        video.style.height = `${videoHeight}px`;
+        video.style.maxHeight = `${videoHeight}px`;
+      } else {
+        video.style.height = `${maxHeight}px`;
+        video.style.maxHeight = `${maxHeight}px`;
+      }
+    });
+  }
+
+  setTimeout(normalizeVideoHeights, 100);
+
```

Comment:
Nit: I was thinking this could be a bit more efficient/simple. Possibly something like this

```suggestion
function normalizeVideoHeights() {
  const videos = el.querySelectorAll('video');
  if (videos.length === 0) return;

  const videoData = [];
  let maxHeight = 0;

  // Collect data and calculate maxHeight
  videos.forEach((video) => {
    const foreground = video.closest('.editorial-card')?.querySelector('.foreground');
    const videoHeight = video.offsetHeight;
    const foregroundHeight = foreground ? foreground.offsetHeight : 0;
    const totalHeight = videoHeight + foregroundHeight;
    videoData.push({ video, foreground, videoHeight, totalHeight });
    if (totalHeight > maxHeight) maxHeight = totalHeight;
  });

  // Apply heights
  videoData.forEach(({ video, foreground, videoHeight }) => {
    if (foreground) {
      video.style.height = `${videoHeight}px`;
      video.style.maxHeight = `${videoHeight}px`;
    } else {
      video.style.height = `${maxHeight}px`;
      video.style.maxHeight = `${maxHeight}px`;
    }
  });
}
```

Reply:
Thanks for the optimization advice!